osquery/events/linux/conntrack.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <sstream>
+
+#include <fnmatch.h>
+#include <linux/limits.h>
+#include <poll.h>
+
+#include <arpa/inet.h>
+
+#include <libmnl/libmnl.h>
+#include <libnetfilter_conntrack/libnetfilter_conntrack.h>
+
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/system.h>
+
+#include "osquery/events/linux/conntrack.h"
+
+namespace osquery {
+
+/// The conntrack subsystem may have a performance impact on the system.
+FLAG(bool,
+     disable_conntrack,
+     true,
+     "Disable receiving events from the conntrack subsystem");
+
+REGISTER(ConntrackEventPublisher, "event_publisher", "conntrack");
+
+Status ConntrackEventPublisher::setUp() {
+  if (FLAGS_disable_conntrack) {
+    return Status(1, "Publisher disabled via configuration");
+  }
+
+  nl_ = std::shared_ptr<struct mnl_socket>(mnl_socket_open(NETLINK_NETFILTER),
+                                           mnl_socket_close);
+  if (nl_ == nullptr) {
+    return Status(1, "Could not open conntrack subsystem");
+  }
+
+  // TODO: How to test if kernel modules have been loaded?
```

Comment:
Still some TODOs, let's create issues for them, implement them, or remove the line.